File: FileCompressionPython/rle_compressor.py
class RLECompressor:
    
    @staticmethod
    def compress(input_text):
        if not input_text:
            return ""
        
        compressed = []
        i = 0
        
        while i < len(input_text):
            current_char = input_text[i]
            count = 1
            
            while i + count < len(input_text) and input_text[i + count] == current_char:
                count += 1
            
            compressed.append(current_char + str(count))
            i += count
        
        return ''.join(compressed)

File: FileCompressionJava/RLECompressor.java
public class RLECompressor {
    
    public static String compress(String input) {
        if (input == null || input.isEmpty()) {
            return "";
        }
        
        StringBuilder compressed = new StringBuilder();
        int i = 0;
        
        while (i < input.length()) {
            char currentChar = input.charAt(i);
            int count = 1;
            
            while (i + count < input.length() && input.charAt(i + count) == currentChar) {
                count++;
            }
            
            compressed.append(currentChar).append(count);
            i += count;
        }
        
        return compressed.toString();
    }
}
